src/main/java/org/radarbase/management/web/rest/SourceDataResource.java

```diff
@@ -71,7 +73,7 @@
     public ResponseEntity<SourceDataDTO> createSourceData(@Valid @RequestBody SourceDataDTO
             sourceDataDto) throws URISyntaxException, NotAuthorizedException {
         log.debug("REST request to save SourceData : {}", sourceDataDto);
-        checkPermission(getJWT(servletRequest), SOURCEDATA_CREATE);
+        checkAuthorityAndPermission(getJWT(servletRequest), SYS_ADMIN, SOURCEDATA_CREATE);
```

Comment:
Just to confirm, would this disregard if the action is performed by a service with client_credentials?

Reply:
Good find. It didn't, but it does now.